ext/hivemq-edge-openapi-2024.8-proposal.yaml

```diff
@@ -0,0 +1,5561 @@
+                      - name: com.hivemq.edge.messages.will.published.count.total
+                      - name: com.hivemq.edge.persistence.client-session.subscriptions.in-memory.total-size
+                      - name: com.hivemq.edge.bridge.simons-cloud.remote.publish.loop-hops-exceeded.count
+                      - name: com.hivemq.edge.networking.bytes.write.total
+                      - name: com.hivemq.edge.bridge.simons-cloud.forward.publish.excluded.count
+                      - name: com.hivemq.edge.networking.connections-closed.total.count
+                      - name: com.hivemq.edge.bridge.simons-cloud.remote.publish.received.count
+                      - name: com.hivemq.edge.subscriptions.overall.current
+                      - name: com.hivemq.edge.persistence.queued-messages.in-memory.total-size
+                      - name: com.hivemq.edge.persistence.client-sessions.in-memory.total-size
+                      - name: com.hivemq.edge.messages.retained.current
+              schema:
+                $ref: '#/components/schemas/MetricList'
+          description: Success
+      summary: Obtain a list of available metrics
+      tags:
+        - Metrics
+        - Metrics Endpoint
+  /api/v1/metrics/{metricName}/latest:
+    get:
+      description: Obtain the latest sample for the metric requested.
+      operationId: getSample
+      parameters:
+        - description: The metric to search for.
+          in: path
+          name: metricName
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          content:
+            application/json:
+              examples:
+                metric-sample:
+                  description: Example response with metrics listed.
+                  summary: Metric Sample
+                  value:
+                    sampleTime: 2023-06-28T11:39:12.789+01
+                    value: 0
+              schema:
+                $ref: '#/components/schemas/DataPoint'
+          description: Success
+      summary: Obtain the latest sample for the metric requested
+      tags:
+        - Metrics
+        - Metrics Endpoint
+  /api/v1/management/domain/tags:
+    get:
+      description: Get the list of all domain tags created in this Edge instance
+      operationId: get-domain-tags
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: string
+          description: Success
+      summary: Get all domain tags
+      tags:
+        - Domain
+  /api/v1/management/domain/tags/schema:
+    get:
+      description: Get the data schema associated with the specified tags
+      operationId: get-tag-schemas
+      parameters:
+        - description: The list of tags to query the schema from
+          in: query
+          name: tags
+          required: true
+          schema:
+            type: array
```

Comment:
Is this really an array of tags? The returned schema is just a single one, so all endpoints need to have the same schema?

Reply:
Good find. The return value should match the params: if it's a list of tags, then it's a list of schemas, indexed by their tags; if it's a single schema, then the param must be a single tag
Will check most common usage and will update